### DISCUSSION TITLE
PHP Undefined Array Key

### DISCUSSION THREAD

**Discussion by twalker-test** [yandeu/five-server#81] [Help / Questions]
I was able to get the PHP and Five servers running. However, I get a PHP error when I click submit on a basic form.

![image](https://user-images.githubusercontent.com/51930083/204205418-ded4943f-39de-4fa1-b7f4-f725e985dd45.png)

My HTML is:

```html
<!DOCTYPE html>
<html>
    <head>
        <title>Form Testing</title>
    </head>

    <body>
        <p>Welcome to the form tester. The goal is to apply simple PHP to save form data.</p>

        <form action="phpbasicform.php" target="_blank" method="post">
            <label for="fname">First Name:</label><br>
            <input type="text" id="fname" name="fname"><br>
            <label for="lname">Last Name:</label><br>
            <input type="text" id="lname" name="lname"><br><br>

            <input type="submit" value="Submit Name or Alias">
        </form>
    </body>
</html>
```

My PHP is:

```php
<html>
    <head>
        <meta charset="UTF-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <script async data-id="five-server" src="http://localhost:5500/fiveserver.js"></script>
        <title>Review</title>
    </head>
    <body>
        Welcome <?php echo $_POST["fname" ]; ?> <?php echo $_POST["lname"]; ?>! <br>
        
        <?php echo "Welcome {$_POST["fname"]} {$_POST["lname"]}!"; ?>
    </body>
</html>
```

When I type in 'localhost:8000', the form works as intended without live updates. I have to save the HTML and PHP files to see it update the information.

Any ideas of what might be causing the error or if I'm using the plugin wrong?

**Comment by yandeu** [comment]
Yes, that's normal. Five-Server does not provide a full PHP Server. It just renders it and displays it in the browser. If you want to use POST request you need a full PHP Server and connect manually to five-server.

See https://github.com/yandeu/five-server-with-php

**Reply by twalker-test** [reply]
The server is now taking POST requests, and the form is functioning correctly. Updates are shown once files are saved. The form refreshes every 3-5secs and removes any information filled out in the form before submitting.

Final Questions:

1. So, was the 'router.php' file not needed, and I only needed the 'header.php' to connect the PHP server to the Five-Server?
2. Can you store any PHP info to be called later (kind of like a database) with the five-server?
3. Is the 'Go Live' button in VSCode for seeing HTML, CSS, and JavaScript with instant updates?

Thanks for all the help!

**Reply by yandeu** [reply]
1) No, the router.php file is not needed. To make it simple, I have removed it from the example.
2) No, there is not such feature in five-server.
3) Yes, but also PHP (render-only), as shown in this video:  
https://youtu.be/4s7q5chX-e0

If you only use PHP during the design process, Five Server for VSCode is a great fit. If you have to test PHP Forms and other "advanced" PHP features, you better use a PHP Server of your choice and connect it manually to five-server (npm), like in the example shown above.

Btw, **there is a way** you can use the VSCode version of Five Server with instant updates with PHP Forms. I just made a video about it:  
https://youtu.be/Qk3WRkzUB88
